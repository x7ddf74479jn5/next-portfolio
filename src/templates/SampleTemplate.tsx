import { GridRowLink, GridRowList, GridRowParagraph } from "src/components/samples/GridRow";
import { WrapInSection } from "src/components/layouts/Section";
import styles from "src/styles/components/samples/SampleGrid.module.scss";

import type { SampleDetail } from "../contents/samples";

type Props = {
  sample: SampleDetail;
};

const SampleTemplate: React.VFC<Props> = ({ sample }) => {
  return (
    <WrapInSection>
      <div className={styles.gridSamples}>
        <img src={sample.img} alt={sample.alt} />
        <GridRowParagraph header={"概要"} data={sample.summary} />
        <GridRowList header={"用途・目的"} data={sample.purposes} />
        <GridRowList header={"こだわり"} data={sample.points} />
        <GridRowParagraph header={"使用言語"} data={sample.langs} />
        <GridRowParagraph header={"技術"} data={sample.techs} />
        <GridRowLink header={"リンク"} data={sample.link} />
        {sample.note && <GridRowParagraph header={"備考"} data={sample.note} />}
      </div>
    </WrapInSection>
  );
};

export default SampleTemplate;
