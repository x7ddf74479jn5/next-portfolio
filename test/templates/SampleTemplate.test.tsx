import SampleTemplate from "src/templates/SampleTemplate";
import type { RenderResult } from "test/test-utils";
import { render } from "test/test-utils";

import type { SampleDetail } from "../../src/contents/samples";

jest.mock("next/image", () => {
  const Image = ({ src, alt }: { src: string; alt: string }): JSX.Element => {
    return <img src={src} alt={alt} />;
  };
  return Image;
});

describe("SampleTemplate", () => {
  let renderResult: RenderResult;
  const testData: SampleDetail = {
    id: "id",
    img: "img.png",
    alt: "alt",
    summary: "summary",
    purposes: ["purpose"],
    points: ["points"],
    langs: "langs",
    techs: "techs",
    link: {
      href: "link",
      label: "label",
      description: "description",
    },
    note: "note",
  };
  const titleArea = {
    title: "title",
    caption: "caption",
  };

  beforeEach(() => {
    renderResult = render(<SampleTemplate sample={testData} titleArea={titleArea} />);
  });

  it("matches snapshot", () => {
    expect(renderResult.asFragment()).toMatchSnapshot();
  });
});
