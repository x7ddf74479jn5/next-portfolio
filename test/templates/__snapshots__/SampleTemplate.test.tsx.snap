// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleTemplate matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="md"
  />
  <section
    class="section"
  >
    <div
      class="sectionTitleArea fadeInLeft"
    >
      <div
        class="headlineSub fadeInUp"
      >
        <h2>
          title
        </h2>
        <p>
          caption
        </p>
        <ul
          aria-label="Breadcrumbs"
          class="textBreadcrumbs"
        >
          <li>
            <span
              aria-current="page"
            >
              HOME
            </span>
             
          </li>
        </ul>
      </div>
    </div>
  </section>
  <section
    class="section"
  >
    <div
      class="sectionWrapIn"
    >
      <div
        class="gridSamples"
      >
        <div
          class="imageWrapper"
        >
          <img
            alt="alt"
            src="img.png"
          />
        </div>
        <div>
          <h3>
            概要
          </h3>
          <p>
            summary
          </p>
        </div>
        <div>
          <h3>
            用途・目的
          </h3>
          <ul>
            <li>
              purpose
            </li>
          </ul>
        </div>
        <div>
          <h3>
            こだわり
          </h3>
          <ul>
            <li>
              points
            </li>
          </ul>
        </div>
        <div>
          <h3>
            使用言語
          </h3>
          <p>
            langs
          </p>
        </div>
        <div>
          <h3>
            技術
          </h3>
          <p>
            techs
          </p>
        </div>
        <div>
          <h3>
            リンク
          </h3>
          <p>
            description
          </p>
          <p>
            <a
              href="link"
              rel="noreferrer"
              target="_blank"
            >
              <span
                class="externalLink"
              >
                label
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  />
                </svg>
              </span>
            </a>
          </p>
        </div>
        <div>
          <h3>
            備考
          </h3>
          <p>
            note
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
